File: airborne_sdk_iOS/hyper-ota/Airborne/AirborneObjC/ApplicationManager/Constants/AJPApplicationConstants.m
//
//  AJPApplicationConstants.m
//  Airborne
//
//  Copyright © Juspay Technologies. All rights reserved.
//

#import <Foundation/Foundation.h>
#import "AJPApplicationConstants.h"

@implementation AJPApplicationConstants

NSString *const JUSPAY_MANIFEST_DIR = @"JuspayManifests";
NSString *const JUSPAY_PACKAGE_DIR = @"JuspayPackages";
NSString *const JUSPAY_RESOURCE_DIR = @"JuspayResources";

NSString *const JUSPAY_MAIN_DIR = @"main";
NSString *const JUSPAY_TEMP_DIR = @"temp";

NSString *const APP_CONFIG_DATA_FILE_NAME = @"app-config.dat";
NSString *const APP_MANIFEST_DATA_TEMP_FILE_NAME = @"app-manifest-temp.dat";

NSString *const APP_PACKAGE_DATA_FILE_NAME = @"app-pkg.dat";
NSString *const APP_PACKAGE_DATA_TEMP_FILE_NAME = @"app-pkg-temp.dat";

NSString *const APP_RESOURCES_DATA_FILE_NAME = @"app-resources.dat";
NSString *const APP_OLD_RESOURCES_DATA_FILE_NAME = @"app-resources-old.dat";
NSString *const APP_TEMP_RESOURCES_DATA_FILE_NAME = @"app-resources-temp.dat";

NSString *const APP_BG_PENDING_DATA_FILE_NAME = @"app-bg-pending.dat";

NSString *const AIRBORNE_NOTIFICATION_TYPE_UPDATE_AVAILABLE = @"UPDATE_AVAILABLE";

NSString *const BOOT_TIMEOUT_NOTIFICATION = @"AJPBootTimeoutNotification";
NSString *const PACKAGE_RESOURCE_NOTIFICATION = @"AJPPackageResourceNotification";
NSString *const RELEASE_CONFIG_NOTIFICATION = @"AJPReleaseConfigNotification";
NSString *const RELEASE_CONFIG_TIMEOUT_NOTIFICATION = @"AJPReleaseConfigTimeoutNotification";
NSString *const LAZY_PACKAGE_NOTIFICATION = @"AJPLazyPackageNotification";

NSString *const APPL_MANAGER_SUB_CAT = @"hyperota";

@end

File: airborne_sdk_iOS/hyper-ota/Airborne/AirborneSwiftModel/AJPApplicationConstants.swift
//
//  AJPApplicationConstants.swift
//  Airborne
//
//  Created by Balaganesh S on 27/04/26.
//


import Foundation

@objcMembers public class AJPApplicationConstants: NSObject {
    // MARK: - Directory Names
    public static let JUSPAY_MANIFEST_DIR = "JuspayManifests"
    public static let JUSPAY_PACKAGE_DIR = "JuspayPackages"
    public static let JUSPAY_RESOURCE_DIR = "JuspayResources"

    public static let JUSPAY_MAIN_DIR = "main"
    public static let JUSPAY_TEMP_DIR = "temp"

    // MARK: - File Names
    public static let APP_CONFIG_DATA_FILE_NAME = "app-config.dat"
    public static let APP_MANIFEST_DATA_TEMP_FILE_NAME = "app-manifest-temp.dat"

    public static let APP_PACKAGE_DATA_FILE_NAME = "app-pkg.dat"
    public static let APP_PACKAGE_DATA_TEMP_FILE_NAME = "app-pkg-temp.dat"

    public static let APP_RESOURCES_DATA_FILE_NAME = "app-resources.dat"
    public static let APP_OLD_RESOURCES_DATA_FILE_NAME = "app-resources-old.dat"
    public static let APP_TEMP_RESOURCES_DATA_FILE_NAME = "app-resources-temp.dat"

    public static let APP_BG_PENDING_DATA_FILE_NAME = "app-bg-pending.dat"

    // MARK: - Notification Names
    public static let BOOT_TIMEOUT_NOTIFICATION = Notification.Name("AJPBootTimeoutNotification")
    public static let PACKAGE_RESOURCE_NOTIFICATION = Notification.Name("AJPPackageResourceNotification")
    public static let RELEASE_CONFIG_NOTIFICATION = Notification.Name("AJPReleaseConfigNotification")
    public static let RELEASE_CONFIG_TIMEOUT_NOTIFICATION = Notification.Name("AJPReleaseConfigTimeoutNotification")
    public static let LAZY_PACKAGE_NOTIFICATION = Notification.Name("AJPLazyPackageNotification")

    // MARK: - Misc
    public static let APPL_MANAGER_SUB_CAT = "hyperota"
}
